app/src/charts/TableChart.tsx

```diff
@@ -7,39 +7,60 @@ import TableContainer from "@material-ui/core/TableContainer";
 import TableHead from "@material-ui/core/TableHead";
 import TableRow from "@material-ui/core/TableRow";
 import { Row } from "../data/DatasetTypes";
-import {
-  getFieldDisplayName,
-  formatFieldValue,
-} from "../utils/madlib/DisplayNames";
 import WarningRoundedIcon from "@material-ui/icons/WarningRounded";
+import { BreakdownVar, BREAKDOWN_VAR_DISPLAY_NAMES } from "../data/Breakdowns";
+import { MetricConfig } from "../data/MetricConfig";
 
-function TableChart(props: { data: Row[]; fields?: string[] }) {
-  const tableColumns: string[] | undefined =
-    !props.fields && props.data.length > 0
-      ? Object.keys(props.data[0])
-      : props.fields;
+// Prints a formatted version of a field value based on the type specified by the field name
+function formatFieldValue(nameOfField: string, value: any): string {
+  if (value === null || value === undefined) {
+    return "";
+  }
+  const formattedValue =
+    typeof value === "number" ? value.toLocaleString("en") : value;
+  const suffix =
+    nameOfField.endsWith("_pct") || nameOfField.endsWith("_pct_of_geo")
+      ? "%"
+      : "";
+  return `${formattedValue}${suffix}`;
+}
```

Comment:
Can we change this to accept MetricType instead of nameOfField and rely on that instead of what the field name ends with? Then we could also move this to the MetricConfig.ts file, since it should be reusable

Reply:
oo yeah good call!